### DISCUSSION TITLE
Exit handler started before workflow was completed.

### DISCUSSION THREAD

**Discussion by IterableTrucks** [argoproj/argo-workflows#12044] [General]
I run hundreds of workflows with  exit handler on workflow's completion every day. There's a chance that one of them starts its exit handler before the last task of the workflow is completed.  My argo workflow version is 3.4.11

**Comment by IterableTrucks** [comment]
I put `workflow.status` variable in the inputs of exit handler task. The `ARGO_TEMPLATE` environment of the task's pod shows `workflow.status` is `Error`. But in the UI, the Inputs/Outputs tab shows the variable is Succeeded

**Reply by agilgur5** [reply]
It's possible that the UI is delayed or inconsistent. Or there's some scenario the UI isn't handling correctly. It listens to an [SSE](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events) Event Stream for changes.

If you refresh the UI does it show it as Error correctly?

**Reply by IterableTrucks** [reply]
No, the UI always shows the status as Succeeded

**Reply by agilgur5** [reply]
What does the completed Workflow CR in k8s say?
As in, are the UI and k8s consistent?

**Reply by IterableTrucks** [reply]
Phase status in UI and k8s workflow resource are consistent.

**Reply by agilgur5** [reply]
Oh, so that would suggest that the exit handler's `workflow.status` is the incorrect one, as it's the odd one out.

Was there an actual error in the Workflow? Is there anything in the Pod's logs? 
Since it only occurs once every few hundred, that suggests a race condition somewhere. I might think that a transient error could perhaps be responsible?
